rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -153,7 +153,7 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
     }
   }
 
-  AnySubscriptionCallback<CallbackMessageT, Alloc> any_callback_;
+  AnySubscriptionCallback<CallbackMessageT, Alloc> & any_callback_;
```

Comment:
@christophebedard I'm somewhat concerned about the lifetime of `SubscriptionIntraProcess` w.r.t `Subscription` and thus `any_callback_` storage. I see we give away references to `SubscriptionIntraProcess` via `get_intra_process_waitable()`:
https://github.com/ros2/rclcpp/blob/b92db52bb14ec097946bfe44d2b337359dddfbab/rclcpp/src/rclcpp/subscription_base.cpp#L179-L196
Then, `SubscriptionIntraProcess` could outlive its associated `Subscription`. Whether that's bad on its own I can't tell. @wjwwood thoughts?

Reply:
@hidmic then, considering what you pointed out, we should probably define/clarify who owns the callback object.

If you'd prefer to leave it as it was before (`SubscriptionIntraProcess` having its own copy of the `AnySubscriptionCallback` object, and not giving it a reference), I can simply copy the tracepoints/registration over to `SubscriptionIntraProcess` so that it takes care of it when intraprocess is enabled.

Reply:
> `SubscriptionIntraProcess` having its own copy of the `AnySubscriptionCallback` object, 

That's probably the safest thing to do. The new intra-process communication architecture is under `experimental` so I'd expect changes to happen and I'd rather not constrain future development defining ownership or banning copies.

Reply:
Got it! I'll submit a different fix soon.

Reply:
@hidmic done. It's over at #918

Reply:
@hidmic done: #918

Reply:
@hidmic done: #918

Reply:
I agree that we should leave it for now and change the tracing logic instead to avoid lifetime issues. I'll close this in favor of the other one.